machine-learning/solutions/seismic_salt_detection/trainer/inputs.py

```diff
@@ -93,10 +119,17 @@ def _input_fn():
         num_threads = multiprocessing.cpu_count()
         dataset = tf.data.TextLineDataset(filenames=[file_path])
         dataset = dataset.skip(1)
-        dataset = dataset.map(lambda x: parse_csv(
-            tf.expand_dims(x, -1)), num_parallel_calls=num_threads)
-        dataset = dataset.map(get_features_target_tuple,
-                              num_parallel_calls=num_threads)
+        dataset = dataset.map(
+            lambda x: parse_csv(tf.expand_dims(x, -1)),
+            num_parallel_calls=num_threads)
+        dataset = dataset.map(
+            get_features_target_tuple,
+            num_parallel_calls=num_threads)
+        dataset = dataset.map(
+            lambda features, target: (
```

Comment:
https://engdoc.corp.google.com/eng/doc/devguide/py/style/index.md#210-lambda-functions says lambdas are ok for "oneliners".  It's super-obnoxious to refactor this, and I understand why you implemented it this way.

I'm not sure what I'd do, myself.  Probably, I'd do all I can to get it to fit on one line.  I think in this case that it's also fine to do nothing.

I thought I'd made this comment last review, but I don't see it now.  If you swap the order of this map and the map before, does the whole thing read clearer?  Then, maybe, you don't need to do any sort of acrobatics.

if you choose to do something such that gpylint still complains, you'll either need to disable that warning now, or if/when this repo is ported to //third_party, you'll need to go back and disable the warning then.

Reply:
This comes from the template:
https://github.com/GoogleCloudPlatform/cloudml-samples/blob/master/cloudml-template/template/trainer/input.py#L227

Not sure what you mean by swapping the order of the maps?  this one needs to happen after  and should not operate in labels (in case labels are not available during production  inference)

Reply:
I understand now. I'll change it. I think the template has it that way in case you need some logic to produce the label that'll be invalid if process_features is ran before.  but in this case this won't happen so I'll change as suggested.

Reply:
(y)

It's tough to reason about what the code should look like when one's scope is larger than just the application/library it's being written for.